cmd/syncv3/main.go

```diff
@@ -113,10 +117,31 @@ func main() {
 		AddPrometheusMetrics: args[EnvPrometheus] != "",
 	})
 
+	if args[EnvSentryDsn] != "" {
+		fmt.Printf("Configuring Sentry reporter...\n")
+		err := sentry.Init(sentry.ClientOptions{
+			Dsn:     args[EnvSentryDsn],
+			Release: version,
+			Dist:    GitCommit,
+		})
+		if err != nil {
+			panic(err)
+		}
+	}
+
 	go h2.StartV2Pollers()
 	if args[EnvJaeger] != "" {
 		h3 = otelhttp.NewHandler(h3, "Sync")
 	}
+
+	if args[EnvSentryDsn] != "" {
+		defer sentry.Flush(2 * time.Second)
```

Comment:
This will never execute because `defer` runs at the end of the function, and this function will never end due to `select {} // block forever`

Reply:
Apparently not. I can crib from Dendrite, e.g. https://github.com/matrix-org/dendrite/blob/236b16aa6c97bc0894388dce7f6b420ef7a1fd88/setup/base/base.go#L524

Reply:
https://github.com/matrix-org/sliding-sync/pull/57/commits/40d9f76b9b64c4a641b85c72c8b534039a19acd0 blocks ditches the infinite select for a SIGINT/TERM